owncloudData/src/test/java/com/owncloud/android/data/sharing/shares/repository/OCShareRepositoryTest.kt

```diff
@@ -19,497 +19,281 @@
+                password = password,
+                expirationDateInMillis = expiration,
+                permissions = permissions,
+                accountName = OC_ACCOUNT_NAME
             )
-        }
-
-        verify(exactly = 1) { localShareDataSource.update(shares.first()) }
-    }
+        } returns OC_PUBLIC_SHARE
 
-    @Test(expected = FileNotFoundException::class)
-    fun updatePublicShareFileNotFoundException() {
         every {
-            remoteShareDataSource.updateShare(
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any()
-            )
-        } throws FileNotFoundException()
+            localShareDataSource.update(OC_PUBLIC_SHARE)
+        } returns 1
 
-        ocShareRepository.updatePublicShare(
-            OC_SHARE.remoteId,
-            "Docs link",
-            "password",
-            2000,
-            -1,
-            accountName
-        )
+        ocShareRepository.updatePublicShare(OC_PUBLIC_SHARE.remoteId, OC_PUBLIC_SHARE.name!!, password, expiration, permissions, OC_ACCOUNT_NAME)
 
         verify(exactly = 1) {
             remoteShareDataSource.updateShare(
-                OC_SHARE.remoteId,
-                "Docs link",
-                "password",
-                2000,
-                -1,
-                accountName
+                remoteId = OC_PUBLIC_SHARE.remoteId,
+                name = OC_PUBLIC_SHARE.name!!,
+                password = password,
+                expirationDateInMillis = expiration,
+                permissions = permissions,
+                accountName = OC_ACCOUNT_NAME
             )
-        }
-
-        verify(exactly = 0) {
-            localShareDataSource.update(shares.first())
+            localShareDataSource.update(OC_PUBLIC_SHARE)
         }
     }
 
     @Test
-    fun insertPrivateShareOk() {
+    fun `getSharesAsLiveData returns a LiveData with a list of OCShares`() {
+        val sharesLiveDataList: LiveData<List<OCShare>> = MutableLiveData(listOfShares)
+
         every {
-            remoteShareDataSource.insert(
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any()
+            localShareDataSource.getSharesAsLiveData(
+                filePath = filePath,
+                accountName = OC_ACCOUNT_NAME,
+                shareTypes = listOf(ShareType.PUBLIC_LINK, ShareType.USER, ShareType.GROUP, ShareType.FEDERATED)
             )
-        } returns shares[2]
+        } returns sharesLiveDataList
 
-        ocShareRepository.insertPrivateShare(
-            filePath,
-            ShareType.GROUP,
-            "whoever",
-            -1,
-            accountName
-        )
+        val sharesResult = ocShareRepository.getSharesAsLiveData(filePath, OC_ACCOUNT_NAME)
+        assertEquals(sharesLiveDataList, sharesResult)
```

Comment:
It's usually better to check the actual content of the `LiveData`. I would recommend returning a list with just 1 share and checking if that share is the one we expect. For that, you'll need the `getLastEmittedValue()` method, which requires adding the `InstantTaskExecutorRule` (check `OCLocalShareDataSourceTest` for an example)